### PR TITLE
build: Few build improvements

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at d8f2070</samp>

This pull request updates various SDK-style projects to use the `TargetFrameworks` property instead of the `TargetFramework` property, to enable multi-targeting for different .NET platforms. It also removes the unnecessary `ToolsVersion` attribute from some project files, and adds some comments to explain the target framework choices for some projects. The changes affect the Lottie, Svg, Xaml, and SourceGenerators libraries, as well as the SamplesApp and the UnoCrossRuntimeLib project template.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.RemoteControl/Uno.UI.RemoteControl.netcoremobile.csproj [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
 	<PropertyGroup>
 		<TargetFrameworks>$(NetCurrentNetCoreMobile);$(NetPreviousNetCoreMobile)</TargetFrameworks>
```
Comment (mrlacey) [inline src/Uno.UI.RemoteControl/Uno.UI.RemoteControl.netcoremobile.csproj:4]: Why no `NetCoreMobilePreviousAndCurrent` - like there is with Skia & WASM?

Comment (Youssef1313) [reply src/Uno.UI.RemoteControl/Uno.UI.RemoteControl.netcoremobile.csproj:4]: Good point. I think I should add this property and use it.

File: src/Uno.UI.Runtime.Skia.Wpf/Uno.UI.Runtime.Skia.Wpf.csproj [hunk 2]
```diff
@@ -1,8 +1,8 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
 	<PropertyGroup>
-		<TargetFrameworks>net7.0-windows</TargetFrameworks>
-		<TargetFrameworks Condition="'$(UnoDisableNet8)'!='true'">$(TargetFrameworks);net8.0-windows</TargetFrameworks>
+		<TargetFrameworks>$(NetPrevious)-windows</TargetFrameworks>
+		<TargetFrameworks Condition="'$(UnoDisableNet8)'!='true'">$(TargetFrameworks);$(NetCurrent)-windows</TargetFrameworks>
```
Comment (mrlacey) [inline src/Uno.UI.Runtime.Skia.Wpf/Uno.UI.Runtime.Skia.Wpf.csproj:5]: Why is this logic here, rather with the other logic about previous and current versions?
Also, this indirectly ties `UnoDisableNet8` and `NetCurrent` - which won't always be the case. - Can this be avoided?
What happens here where NetCurrent isn't the same as Net8?

File: src/Uno.UI.XamlHost.Skia.Wpf/Uno.UI.XamlHost.Skia.Wpf.csproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="MSBuild.Sdk.Extras">
 
 	<PropertyGroup>
-		<TargetFrameworks>net7.0-windows;net8.0-windows</TargetFrameworks>
+		<TargetFrameworks>$(NetPrevious)-windows;$(NetCurrent)-windows</TargetFrameworks>
```
Comment (mrlacey) [inline src/Uno.UI.XamlHost.Skia.Wpf/Uno.UI.XamlHost.Skia.Wpf.csproj:4]: Why no `NetWindowsPreviousAndCurrent` - like there is with Skia & WASM?

**Review by mrlacey — APPROVED** [review]
approved subject to comments

File: src/Uno.UI.Runtime.Skia.Wpf/Uno.UI.Runtime.Skia.Wpf.csproj (hunk 2 above)
Comment (Youssef1313) [reply src/Uno.UI.Runtime.Skia.Wpf/Uno.UI.Runtime.Skia.Wpf.csproj:5]: I think the property should be named `UnoDisableNetCurrent`

This part is essentially a workaround for a long-standing .NET SDK bug (https://github.com/dotnet/sdk/issues/30103)

File: src/Uno.UI.XamlHost.Skia.Wpf/Uno.UI.XamlHost.Skia.Wpf.csproj (hunk 3 above)
Comment (Youssef1313) [reply src/Uno.UI.XamlHost.Skia.Wpf/Uno.UI.XamlHost.Skia.Wpf.csproj:4]: I should add this property, but I think having `Wpf` in the name would be clearer to not confuse it with UWP/WinUI TFMs (uap and netX.0-windows10.0.xxxx)

**Comment by Youssef1313** [comment]
@mrlacey Can you take another look? Let me know if you have any more comments.

File: .github/workflows/winui-autoconvert.yml [hunk 4]
```diff
@@ -11,8 +11,8 @@ env:
   UnoCheck_Manifest: 'https://raw.githubusercontent.com/unoplatform/uno.check/146b0b4b23d866bef455494a12ad7ffd2f6f2d20/manifests/uno.ui.manifest.json'
 
   # Only convert using net7
```
Comment (Youssef1313) [inline .github/workflows/winui-autoconvert.yml:13]:
```suggestion
  # Only convert using NetPrevious
```

File: build/ci/.azure-devops-android-tests.yml [hunk 5]
```diff
@@ -30,8 +30,8 @@ jobs:
     NUGET_PACKAGES: $(Agent.WorkFolder)/.nuget
 
     # Only build for net7
```
Comment (Youssef1313) [inline build/ci/.azure-devops-android-tests.yml:32]:
```suggestion
    # Only build for NetPrevious
```

File: build/ci/.azure-devops-ios-tests.yml [hunk 6]
```diff
@@ -54,8 +54,8 @@ jobs:
     NUGET_PACKAGES: $(Pipeline.Workspace)/.nuget/packages
 
     # Only build for net7
```
Comment (Youssef1313) [inline build/ci/.azure-devops-ios-tests.yml:56]:
```suggestion
    # Only build for NetPrevious
```

File: build/ci/.azure-devops-skia-tests.yml [hunk 7]
```diff
@@ -20,8 +20,8 @@ jobs:
     NUGET_PACKAGES: $(Agent.WorkFolder)\.nuget
 
     # Only build for net7
```
Comment (Youssef1313) [inline build/ci/.azure-devops-skia-tests.yml:22]:
```suggestion
    # Only build for NetPrevious
```

File: build/ci/.azure-devops-uap.yml [hunk 8]
```diff
@@ -16,8 +16,8 @@ jobs:
     NUGET_PACKAGES: $(Agent.WorkFolder)\.nuget
 
     # Only build for net7
```
Comment (Youssef1313) [inline build/ci/.azure-devops-uap.yml:18]:
```suggestion
    # Only build for NetPrevious
```

File: build/ci/.azure-devops-unit-tests.yml [hunk 9]
```diff
@@ -36,8 +36,8 @@ jobs:
     NUGET_PACKAGES: $(Agent.WorkFolder)\.nuget
 
     # Only build for net7
```
Comment (Youssef1313) [inline build/ci/.azure-devops-unit-tests.yml:38]:
```suggestion
    # Only build for NetPrevious
```

File: build/ci/.azure-devops-unit-tests.yml [hunk 10]
```diff
@@ -118,8 +118,8 @@ jobs:
...
```
Comment (Youssef1313) [inline build/ci/.azure-devops-unit-tests.yml:120]:
```suggestion
    # Only build for NetPrevious
```

File: build/ci/.azure-devops-wasm-uitests.yml [hunk 11]
```diff
@@ -13,8 +13,8 @@ jobs:
     NUGET_PACKAGES: $(build.sourcesdirectory)/.nuget
 
     # Only build for net7
```
Comment (Youssef1313) [inline build/ci/.azure-devops-wasm-uitests.yml:15]:
```suggestion
    # Only build for NetPrevious
```

File: build/ci/.azure-devops-winui-convert.yml [hunk 12]
```diff
@@ -18,8 +18,8 @@ jobs:
     XAML_FLAVOR_BUILD: WinUI
 
     # Only build for net7
```
Comment (Youssef1313) [inline build/ci/.azure-devops-winui-convert.yml:20]:
```suggestion
    # Only build for NetPrevious
```

**Review by mrlacey — APPROVED** [review]
LGTM

File: src/SamplesApp/UnoIslandsSamplesApp.Skia.Wpf/UnoIslandsSamplesApp.Skia.Wpf.csproj [hunk 13]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="MSBuild.Sdk.Extras">
 	<PropertyGroup>
-		<TargetFrameworks>$(NetPrevious)-windows;$(NetCurrent)-windows</TargetFrameworks>
+		<TargetFrameworks>NetWpfPreviousAndCurrent</TargetFrameworks>
```
Comment (Youssef1313) [inline src/SamplesApp/UnoIslandsSamplesApp.Skia.Wpf/UnoIslandsSamplesApp.Skia.Wpf.csproj:3]:
```suggestion
		<TargetFrameworks>$(NetWpfPreviousAndCurrent)</TargetFrameworks>
```

File: src/Uno.UI.XamlHost.Skia.Wpf/Uno.UI.XamlHost.Skia.Wpf.csproj [hunk 14]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="MSBuild.Sdk.Extras">
 
 	<PropertyGroup>
-		<TargetFrameworks>net7.0-windows;net8.0-windows</TargetFrameworks>
+		<TargetFrameworks>NetWpfPreviousAndCurrent</TargetFrameworks>
```
Comment (Youssef1313) [inline src/Uno.UI.XamlHost.Skia.Wpf/Uno.UI.XamlHost.Skia.Wpf.csproj:4]:
```suggestion
		<TargetFrameworks>$(NetWpfPreviousAndCurrent)</TargetFrameworks>
```

File: src/Uno.UI.RemoteControl.VS/Uno.UI.RemoteControl.VS.csproj [hunk 15]
```diff
@@ -1,6 +1,7 @@
-<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
+<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
+	<!-- TODO: Figure out if we really need to multi-target this or if net48 is enough -->
```
Comment (jeromelaban) [inline src/Uno.UI.RemoteControl.VS/Uno.UI.RemoteControl.VS.csproj:4]: net461 is for vs2019.

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI.RemoteControl.VS/Uno.UI.RemoteControl.VS.csproj (hunk 15 above)
Comment (Youssef1313) [reply src/Uno.UI.RemoteControl.VS/Uno.UI.RemoteControl.VS.csproj:4]: @jeromelaban But do we still care about VS 2019?

Comment (jeromelaban) [reply src/Uno.UI.RemoteControl.VS/Uno.UI.RemoteControl.VS.csproj:4]: For now yes, I should have mentioned. Let's keep it until it causes us build trouble.

**Pull request merged**